staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -403,14 +404,17 @@ func (r *Request) RequestURI(uri string) *Request {
 		r.err = err
 		return r
 	}
-	r.pathPrefix = locator.Path
+	// AbsPath handles prepending r.c.base.Path, if set
+	r.AbsPath(locator.Path)
 	if len(locator.Query()) > 0 {
-		if r.params == nil {
-			r.params = make(url.Values)
-		}
+		// clear any existing params
+		r.params = make(url.Values)
```

Comment:
before we were appending the new params to the existing ones?

Reply:
yeah, it was a mess

Reply:
RequestURI("/foo?bar=baz") could end up issuing requests for the specified path with random other previously specified params still lingering